templates/experiments/single_experiment_home.html

```diff
@@ -389,4 +532,238 @@ <h3 class="font-bold text-lg">Parent Experiments:
+              const filter = filters[i];
+              filter.column = columnParam;
+
+              if (this.filterData.columns[columnParam]) {
+                filter.availableOperators = this.filterData.columns[columnParam].operators;
+                filter.operator = operatorParam;
+
+                if (['any of', 'all of'].includes(operatorParam) && valueParam) {
+                  try {
+                    filter.selectedValues = JSON.parse(valueParam);
+                    if (this.filterData.columns[columnParam].options) {
+                      filter.filteredOptions = [...this.filterData.columns[columnParam].options];
+                    }
+                  } catch (e) {
+                    console.error('Error parsing filter value from URL:', e);
+                  }
+                } else if (valueParam) {
+                  filter.value = valueParam;
+                }
+              }
+            }
+          }
+          this.filterData.filters = filters.filter(filter =>
+            filter.column && filter.operator && (
+              filter.value ||
+              (Array.isArray(filter.selectedValues) && filter.selectedValues.length > 0)
+            )
+          );
+          if (this.filterData.filters.length > 0) {
+            this.filterData.showFilters = true;
+          } else {
+            this.addFilter();
+          }
+        },
+
+        updateUrlWithFilters() {
+          const activeFilters = this.filterData.filters.filter(filter =>
+            filter.column && filter.operator && (
+              filter.value ||
+              (Array.isArray(filter.selectedValues) && filter.selectedValues.length > 0)
+            )
+          );
+          const params = new URLSearchParams(window.location.search);
+          [...params.keys()].forEach(key => {
+            if (key.startsWith('filter_')) {
+              params.delete(key);
+            }
+          });
+          activeFilters.forEach((filter, i) => {
+            params.set(`filter_${i}_column`, filter.column);
+            params.set(`filter_${i}_operator`, filter.operator);
+
+            if (['any of', 'all of'].includes(filter.operator)) {
+              if (filter.selectedValues && filter.selectedValues.length > 0) {
+                params.set(`filter_${i}_value`, JSON.stringify(filter.selectedValues));
+              }
+            } else {
+              params.set(`filter_${i}_value`, filter.value);
+            }
+          });
+
+          const showAllCheckbox = document.querySelector('input[name="show-all"]');
+          if (showAllCheckbox && showAllCheckbox.checked) {
+            params.set('show-all', 'on');
+          } else {
+            params.delete('show-all');
+          }
+
+          const newUrl = window.location.pathname + (params.toString() ? '?' + params.toString() : '') + window.location.hash;
+          window.history.replaceState({}, '', newUrl);
+        },
+
+        triggerFilterChange(updateUrl = true) {
```

Comment:
When adding a new filter and selecting the column etc, this gets called and makes a request even though the filter isn't complete yet. I'm not sure the best way to fix that but one option might be to pass the index of the filter being edited and check if it's valid before continuing.